File: Android/Reddit/app/src/main/java/com/ognjenlazic/reddit/model/Community.kt
package com.ognjenlazic.reddit.model

data class Community(
    val communityId: Int,
    val creationDate: Long,
    var description: String,
    val moderatorId: Int,
    var name: String,
    val rules: List<Rule>?,
    val suspended: Boolean,
    val suspendedReason: Any?
){
    constructor(): this(
        0, 0, "", 0, "", emptyList(), false, ""
    )
}

File: Frontend/reddit/src/app/model/community.ts
import { Rule } from 'src/app/model/rule';

export interface Community {
    communityId: number;
    name: string;
    description: string;
    creationDate: Date;
    isSuspended: boolean;
    suspendedReason: string;
    rules : Rule[] | null;
    moderatorId: number, 
    
    // moderators: [key:number, value: User];
    // flairs: [key:number, value: Falir];
    // rules: [key:number, value: Rule];
    // posts: [key:number, value: Post];
}